src/structures/Message.js

```diff
@@ -596,13 +596,13 @@ class Message extends Base {
    */
   get crosspostable() {
     return (
-      this.channel.type === 'GUILD_NEWS' &&
+      this.channel?.type === 'GUILD_NEWS' &&
       !this.flags.has(MessageFlags.FLAGS.CROSSPOSTED) &&
       this.type === 'DEFAULT' &&
-      this.channel.viewable &&
-      this.channel.permissionsFor(this.client.user).has(Permissions.FLAGS.SEND_MESSAGES) &&
+      this.channel?.viewable &&
```

Comment:
there isno need to use optional chaining here, above `this.channel?.type === '...' && ...` would ensure channel exists

*unrelated but since `this.channel` is a getter now should it be declared as a variable first instead of repeated references*
```js
const { channel } = this;
return ...
```

Reply:
I forgot that it would stop after one of the conditions returns false, will revert that part then

Reply:
As for the const channel, is it really needed? I mean these are just boolean checks so they should all be running at the same time. Also not sure if it's within the scope of this PR but you/the maintainers could tell me

Reply:
yeah it is unrelated to this pr, just thought id mention it

since `this.channel` is a getter im assuming that the getter function is ran every time its referenced

so its essentially like

`this.client.channels.resolve(...)?.type === ... && this.client.channels.resolve(...)... && ...
```
or maybe im overthinking it and this isn't an issue, idk